File: FinBalancer.Api/Models/AuthResult.cs
namespace FinBalancer.Api.Models;

public class AuthResult
{
    public bool Success { get; set; }
    public string? Token { get; set; }
    /// <summary>Refresh token - valid 1 year. Use to obtain new access token.</summary>
    public string? RefreshToken { get; set; }
    public DateTime? TokenExpiresAt { get; set; }
    public User? User { get; set; }
    public string? Error { get; set; }
}

File: FinBalancer.App/lib/models/auth_result.dart
import 'user.dart';

class AuthResult {
  final bool success;
  final String? token;
  final User? user;
  final String? error;

  AuthResult({
    required this.success,
    this.token,
    this.user,
    this.error,
  });

  factory AuthResult.fromJson(Map<String, dynamic> json) {
    return AuthResult(
      success: json['success'] as bool? ?? false,
      token: json['token'] as String?,
      user: json['user'] != null ? User.fromJson(json['user'] as Map<String, dynamic>) : null,
      error: json['error'] as String?,
    );
  }
}
